src/main/java/io/cdap/plugin/servicenow/source/ServiceNowMultiSourceConfig.java

```diff
@@ -0,0 +1,140 @@
+                                     String valueType, @Nullable String startDate, @Nullable String endDate,
+                                     String tableNames) {
+    super(referenceName, tableNameField, clientId, clientSecret, restApiEndpoint, user, password, valueType, startDate,
+      endDate);
+    this.tableNames = tableNames;
+  }
+
+  @Nullable
+  public String getTableNames() {
+    return tableNames;
+  }
+
+  /**
+   * Validates {@link ServiceNowMultiSourceConfig} instance.
+   */
+  public void validate(FailureCollector collector) {
+    // Validates the given referenceName to consists of characters allowed to represent a dataset.
+    IdUtils.validateReferenceName(referenceName, collector);
+
+    validateCredentials(collector);
+    validateTableNames(collector);
+    validateValueType(collector);
+    validateDateRange(collector);
+    validateTableNameField(collector);
+  }
+
+  public void validateTableNames(FailureCollector collector) {
+    if (containsMacro(ServiceNowConstants.PROPERTY_TABLE_NAMES)) {
+      return;
+    }
+
+    if (Util.isNullOrEmpty(tableNames)) {
+      collector.addFailure("Table names must be specified.", null)
+        .withConfigProperty(ServiceNowConstants.PROPERTY_TABLE_NAMES);
+    } else {
+      Set<String> tableNames = ServiceNowMultiInputFormat.getList(getTableNames());
+      for (String tableName : tableNames) {
+        // Call API to fetch first record from the table
+        ServiceNowTableAPIRequestBuilder requestBuilder = new ServiceNowTableAPIRequestBuilder(
+          this.getRestApiEndpoint(), tableName)
+          .setExcludeReferenceLink(true)
+          .setDisplayValue(this.getValueType())
+          .setLimit(1);
+
+        RestAPIResponse apiResponse = null;
+        ServiceNowTableAPIClientImpl serviceNowTableAPIClient = new ServiceNowTableAPIClientImpl(this);
+        try {
+          String accessToken = serviceNowTableAPIClient.getAccessToken();
+          requestBuilder.setAuthHeader(accessToken);
+
+          // Get the response JSON and fetch the header X-Total-Count. Set the value to recordCount
+          requestBuilder.setResponseHeaders(HEADER_NAME_TOTAL_COUNT);
+
+          apiResponse = serviceNowTableAPIClient.executeGet(requestBuilder.build());
+          if (!apiResponse.isSuccess()) {
+            if (apiResponse.getHttpStatus() == HttpStatus.SC_BAD_REQUEST) {
+              collector.addFailure("Bad Request. Table: " + tableName + " is invalid.", "")
+                .withConfigProperty(ServiceNowConstants.PROPERTY_TABLE_NAMES);
+            }
+          } else if (serviceNowTableAPIClient.parseResponseToResultListOfMap(apiResponse.getResponseBody()).isEmpty()) {
+            collector.addFailure("Table: " + tableName + " is empty.", "")
+              .withConfigProperty(ServiceNowConstants.PROPERTY_TABLE_NAMES);
+          }
+        } catch (OAuthSystemException | OAuthProblemException e) {
+          LOG.error("Error in fetchTableRecords", e);
```

Comment:
add errors in failure collector

Reply:
Errors added to the failure collector 7661e08